ImperatorToCK3/Source/CK3/Province/ProvinceDetails.cpp

```diff
@@ -1,42 +1,30 @@
 #include "ProvinceDetails.h"
 #include "Log.h"
 #include "OSCompatibilityLayer.h"
-#include "ParserHelpers.h"
 #include "CommonRegexes.h"
 
-CK3::ProvinceDetails::ProvinceDetails(const std::string& filePath)
-{
-	registerKeys();
-	if (commonItems::DoesFileExist(filePath))
-	{
-		parseFile(filePath);
-	}
-	else Log(LogLevel::Error) << "Could not open " << filePath << " to load province details.";
-	clearRegisteredKeywords();
-}
 
-void CK3::ProvinceDetails::updateWith(const std::string& filePath)
-{
-	registerKeys();
-	if (commonItems::DoesFileExist(filePath))
-	{
-		parseFile(filePath);
-	}
-	else Log(LogLevel::Error) << "Could not open " << filePath << " to update province details.";
-	clearRegisteredKeywords();
-}
 
-CK3::ProvinceDetails::ProvinceDetails(std::istream& theStream)
-{
-	registerKeys();
-	parseStream(theStream);
-	clearRegisteredKeywords();
-}
+History::Factory CK3::ProvinceDetails::historyFactory = History::Factory({
+	{"culture", "culture", std::nullopt},
```

Comment:
It would be helpful to insert comments describing what each of these set:

```
{/* keyword */ "culture", /* field name */ "culture", /* default value */ std::nullopt}
```

Reply:
Alternately, use the new aggregate initialization form (https://en.cppreference.com/w/cpp/language/aggregate_initialization)

```
{.fieldName="culture", .setter="culture", .initialValue=std::nullopt},
```

In either case, it makes it easier to read the code and see what's being set.

Reply:
Thanks, used aggregate initialization.